osu.Game/Overlays/BeatmapSet/LeaderboardScopeSelector.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.UserInterface;
+using osu.Game.Screens.Select.Leaderboards;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osuTK;
+using System;
+using osu.Game.Graphics.UserInterface;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics.Sprites;
+using osu.Framework.Extensions;
+using osu.Game.Graphics;
+using osu.Framework.Allocation;
+using osuTK.Graphics;
+using osu.Framework.Input.Events;
+using osu.Framework.Graphics.Colour;
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public class LeaderboardScopeSelector : TabControl<BeatmapLeaderboardScope>
+    {
+        protected override Dropdown<BeatmapLeaderboardScope> CreateDropdown() => null;
+
+        protected override TabItem<BeatmapLeaderboardScope> CreateTabItem(BeatmapLeaderboardScope value) => new ScopeSelectorTabItem(value);
+
+        public LeaderboardScopeSelector()
+        {
+            AutoSizeAxes = Axes.Y;
+            RelativeSizeAxes = Axes.X;
+
+            foreach (var scope in Enum.GetValues(typeof(BeatmapLeaderboardScope)))
+            {
+                if (scope is BeatmapLeaderboardScope.Local)
+                    continue;
+
+                AddItem((BeatmapLeaderboardScope)scope);
+            }
+
+            AddInternal(new Line
+            {
+                Anchor = Anchor.BottomCentre,
+                Origin = Anchor.BottomCentre,
+            });
+        }
+
+        protected override TabFillFlowContainer CreateTabFlow() => new TabFillFlowContainer
+        {
+            Anchor = Anchor.TopCentre,
+            Origin = Anchor.TopCentre,
+            AutoSizeAxes = Axes.Both,
+            Direction = FillDirection.Horizontal,
+            Spacing = new Vector2(20, 0),
+        };
+
+        private class ScopeSelectorTabItem : TabItem<BeatmapLeaderboardScope>
+        {
+            private const float transition_duration = 100;
+
+            private readonly Box box;
+
+            protected readonly OsuSpriteText Text;
+            protected readonly OsuSpriteText BoldText;
+
+            public ScopeSelectorTabItem(BeatmapLeaderboardScope value)
+                : base(value)
+            {
+                AutoSizeAxes = Axes.Both;
+
+                Children = new Drawable[]
+                {
+                    Text = new OsuSpriteText
+                    {
+                        Margin = new MarginPadding { Bottom = 8 },
+                        Origin = Anchor.BottomCentre,
+                        Anchor = Anchor.BottomCentre,
+                        Text = value.GetDescription() + "Ranking",
+                        Font = OsuFont.GetFont(weight: FontWeight.Light),
+                        AlwaysPresent = true,
+                    },
+                    BoldText = new OsuSpriteText
```

Comment:
Totally unnecessary, you can bold the normal text by doing `Text.Font = Text.Font.With(weight: FontWeight.Black),`

Reply:
not in this case. I'm using autosized FillFlowContainer. Using your suggestion will broke all the layout (buttons will jump on hovering)

Reply:
This is a really ugly and performance affecting way of doing thing :/

Reply:
I'll just use fixed width then

Reply:
Have you checked how the web does it, out of curiosity?

Reply:
oh, okay, got it